Gordon360/Controllers/ProfilesController.cs

```diff
@@ -449,6 +450,37 @@ public async Task<ActionResult<StudentProfileViewModel>> UpdateMobilePhoneNumber
             return Ok(result);
         }
 
+        /// <summary>
+        /// Update planned graduation year
+        /// </summary>
+        /// <param name="value">planned graduation year</param>
+        /// <returns></returns>
+        [HttpPut]
+        [Route("planned_graduation_year")]
+        //
+        public async Task<ActionResult<StudentProfileViewModel>> UpdatePlannedGraduationYearAsync(string value,string username)
+        {
+            //var username = AuthUtils.GetUsername(User);
+            var result = await _profileService.UpdatePlannedGraduationYearAsync(username, value);
+
+            return Ok(result);
+        }
+
+        /// <summary>
+        /// Update privacy of planned graduation year
+        /// </summary>
+        /// <param name="value">Y or N</param>
+        /// <returns></returns>
+        [HttpPut]
+        [Route("planned_graduation_year_privacy/{value}")]
```

Comment:
Yeah, that makes sense, but before we have the UserPrivacy table and its frond and back end fully working, would it be good that we can store this privacy in Custom_Profile table for now?

Reply:
That's true! I'll work on it

Reply:
By the way Evan, before the UserPrivacy table and its frond and back end working, do you think if it's okay that we always have the planned graduation year be public?

Reply:
That's probably a question for @ChrisImagineer. Chris, the practicum students are adding a "new" field to profiles to show a student's self-declared planned graduation year. Is it acceptable for that to always be public, at least for now? My instinct is yes, because the data will only exist if the student adds it themselves.

Reply:
Actually, after talking with Dr. Tuck, we're wondering if the planned graduation year would be only editable by the advisors. Or, students can edit them, but they need their advisor's permission (like a email automatically sent to the advisors).

Reply:
for now we are always showing them in the UI (public), once the table of user_privacy is changed we can use this route to change that table.